specs/AllowHostInputProcessing.md

```diff
@@ -0,0 +1,101 @@
+# Background
+WebView2 .NET developers often encounter issues where certain default events are not raised in the WebView2 .NET control. 
+
+This issue occurs because user inputs are directly delivered to the browser, and the host application does not receive the corresponding messages. 
+
+This causes some default event handling API of .NET control not to work, such as PreProcessMessage and ProcessCmdKey in WinForms.
+
+It also prevents some keys from reaching Key Event realated APIs such as OnKeyDown and ToolStripMenuItem.
+
+In order to solve this type of issue, we provide an api which allows user inputs pass through the browser to the host app.
+
+# Description
+`AllowHostInputProcessing` allows user input messages to pass through the browser window to be received by an app process window.
```

Comment:
Does this mean that input messages are IPC'd from the browser process to the host process, and then IPC'd back from the host process to the browser process afterward? This sounds like a big performance hit, and may also create weird reentrancy bugs.

Reply:
No, the input messages are directly delivered into host app process and expected to be dispatched to HWND owned by WebView2 Controller. Then the input message will be IPC'd to the browser process.